### ISSUE TITLE
image preprocessing

### ISSUE THREAD

**Issue by bakachan19** [Docta-ai/docta#3]
Hi. 
Thank you for this interesting package.

I was looking at the examples in the ```tools```, more precisely at ```docta_rare_pattern.py``` and using it with the following command:
```%run ./tools/docta_rare_pattern.py --feature_type 'embedding' --suffix 'c1m_subset' ```, following the demo in the notebook [docta_rare_pattern_clothes.ipynb](https://github.com/Docta-ai/docta/blob/master/demo/docta_rare_pattern_clothes.ipynb).

In this case the dataset is created using:
```dataset = Customize_Image_Folder(root=cfg.data_root, transform=None)```.

Looking at the  ```Customize_Image_Folder()```, given that ```transform =None``` the image will be transformed using the following function:
```  
if transform is None:
            self.transform = transforms.Compose([
                transforms.Resize(256),
                transforms.CenterCrop(224),
                transforms.ToTensor(),
                transforms.Normalize((0.6959, 0.6537, 0.6371), (0.3113, 0.3192, 0.3214)),
            ])
```
and followed up by:

```
feature_all.append(tmp_feature.permute(1, 2, 0).numpy().astype(np.uint8))  
```

But then in the ```docta_rare_pattern.py``` file the ```pre_processor.encode_feature()``` is used which customized the data (performs another normalization step) with:
```dataset_list += [CustomizedDataset(feature=self.dataset.feature, label=self.dataset.label, preprocess=preprocess)]```, where the preprocess function is retrieved from ```get_encoder()```:
```model_embedding, _, preprocess = open_clip.create_model_and_transforms(self.cfg.embedding_model)```

I am wondering why the images are normalized two times: one within the ```Customized_Image_Folder()``` and a second time with the preprocessing function from ```open_clip```.
Are there any steps that I am missing? I cannot fully understand why the double normalization&preprocessing.
Could you please explain why this strategy is used in the method?

I am greatly thankful for any hints and insights that you can provide.
Thank you for your time.

**Comment by weijiaheng** [comment]
Hi

Thanks for the great catching and your detailed post! We have fixed this "double processing" in our latest commit.

Briefly speaking, in the file ``customize_img_folder.py``, we removed the data preprocessing code when ``transform=None`` to avoid double preprocessing. In this case, Docta will simply append features by changing each image from ```<PIL.Image.Image>``` to the numpy array. Thus, the revised code will process & normalize the image data for only once when ``transform=None`` (default value).

Please feel free to let us know if you have any additional concerns!

Best,
Jiaheng

**Comment by bakachan19** [comment]
Hi @weijiaheng.
Thank you for your reply and for helping me clear my doubts.

I also went over again the CIFAR10 preprocessing cause I thought there was the same double processing issue also there.
But I missed the following:
https://github.com/Docta-ai/docta/blob/100384d8149d8bae0a172efed808751e4cc1d89f/docta/datasets/cifar.py#L33

and here ```self.data``` is not normalized yet (since the normalization is applied by the dataloader and not by the dataset)
and then it just undergoes the preprocessing of CLIP with:
```model_embedding, _, preprocess = open_clip.create_model_and_transforms(self.cfg.embedding_model)```
and 
```CustomizedDataset(..., preprocess = preprocess)``` in line: https://github.com/Docta-ai/docta/blob/100384d8149d8bae0a172efed808751e4cc1d89f/docta/core/preprocess.py#L129

Thank you once again for your help.
Have an amazing day.

**Comment by zwzhu-d** [comment]
Yes, the CIFAR data is not double-processed. The transforms in the CIFAR dataloader are left for future extensions. Thank you for the good catching.

**Comment by bakachan19** [comment]
Thank you for the clarification @zwzhu-d and @weijiaheng.
It was extremely useful.
